### PR TITLE
Clean up the IDE

### PR DESCRIPTION
**Pull request by neil-da**
More work spring cleaning the IDE stuff. In particular:

* Rename things that clashed
* Make a proper Options module
* Reduce the number of logging things that weren't used

### CODE REVIEW

File: compiler/haskell-ide-core/src/Development/IDE/Functions/Compile.hs [hunk 1]
```diff
@@ -61,24 +60,6 @@ import           Development.IDE.Types.SpanInfo
 import GHC.Generics (Generic)
 import           System.FilePath
 
--- TODO (MK) Move to a separate Options module
```
Comment (cocreature) [inline compiler/haskell-ide-core/src/Development/IDE/Functions/Compile.hs]: :+1:

File: compiler/haskell-ide-core/src/Development/IDE/State/FileStore.hs [hunk 2]
```diff
@@ -141,7 +142,13 @@ setBufferModified state absFile (mcontents, !time) = do
...
-        Just contents -> Map.insert absFile $ strictPair time (textToStringBuffer contents)
+        Just contents -> Map.insert absFile $ strictPair time $ textToStringBuffer contents
 
     -- run shake to update results regarding the files of interest
     void $ shakeRun state []
+
+
+-- would be nice to do this more efficiently...
+textToStringBuffer :: T.Text -> SB.StringBuffer
+-- would be nice to do this more efficiently...
```
Comment (hurryabit) [inline compiler/haskell-ide-core/src/Development/IDE/State/FileStore.hs]: You must be very worried about efficiency here. ;)

Comment (neil-da) [reply compiler/haskell-ide-core/src/Development/IDE/State/FileStore.hs]: 🤦‍♂

Comment (neil-da) [reply compiler/haskell-ide-core/src/Development/IDE/State/FileStore.hs]: Removed in https://github.com/digital-asset/daml/pull/963/files#diff-b8ba0a4cf59ff60c866c80c7ccb8f198

**Pull request merged**